### PR TITLE
{bio}[goolf/1.4.10] CONTRAlign 2.01 (REVIEW)

### PR DESCRIPTION
**Pull request by pescobar**
added the two possible "flavours" of this app and the required patch for gcc-4.7.2

### CODE REVIEW

**Comment by fgeorgatos** [comment]
Test report by @fgeorgatos
**SUCCESS**
Build succeeded for 2 out of 2
Linux debian 6.0.10, Intel(R) Xeon(R) CPU           L5640  @ 2.27GHz, Python 2.6.6
See https://gist.github.com/81483ad50b6f645c416e for a full test report.

File: easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-proteins-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+easyblock = 'MakeCp'
+
+homepage = 'http://contra.stanford.edu/contralign/'
+description = ''' CONditional TRAining for Protein Sequence Alignment  '''
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'}
+
+source_urls = [homepage]
+sources = ['%s_v%s.tar.gz' % (name.lower(),version.replace('.','_').strip('-proteins'))]
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-proteins-goolf-1.4.10.eb]: @pescobar: plz follow up? this is not difficult to tune

**Comment by fgeorgatos** [comment]
minor changes needed here, put this in REVIEW plz!

**Comment by boegel** [comment]
@pescobar: will send you a PR to fix these remarks

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-rna-goolf-1.4.10.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = [homepage]
+sources = ['%%(namelower)s_v%s.tar.gz' % version.replace('.', '_')]
+
+patches = ['CONTRAlign-%(version)s_gcc47.patch']
+
+start_dir = 'src'
+
+# this option determines whether the program is being compiled for protein or RNA sequence input. 
+# Here we are compiling for RNA input. Check docs for details
+makeopts = 'MODEL_TYPE="-DRNA=1"'
```
Comment (boegel) [inline easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-rna-goolf-1.4.10.eb]: should be `buildopts`

File: easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-proteins-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = [homepage]
+sources = ['%%(namelower)s_v%s.tar.gz' % version.replace('.', '_')]
+
+patches = ['CONTRAlign-%(version)s_gcc47.patch']
+
+start_dir = 'src'
+
+# this option determines whether the program is being compiled for protein or RNA sequence input. 
+# Here we are compiling for protein input. Check docs for details
+makeopts = 'MODEL_TYPE="-DRNA=0"'
```
Comment (boegel) [inline easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01-proteins-goolf-1.4.10.eb]: `buildopts`

**Comment by boegel** [comment]
@pescobar ping on remarks?

**Comment by pescobar** [comment]
Test report by @pescobar
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
login18.cluster.bc2.ch - Linux centos 6.5, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.6.6
See https://gist.github.com/b1ba1324fc6289b1cdf790a2299f0ffa for a full test report.

**Comment by boegel** [comment]
@pescobar filename should be `CONTRAlign-2.01-goolf-1.4.10-proteins.eb` and `CONTRAlign-2.01-goolf-1.4.10-rna.eb`, see Travis errors

File: easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01_gcc47.patch [hunk 4]
```diff
@@ -0,0 +1,21 @@
+--- contralign.orig/src/Makefile	2008-08-15 03:06:30.000000000 +0200
```
Comment (boegel) [inline easybuild/easyconfigs/c/CONTRAlign/CONTRAlign-2.01_gcc47.patch]: @pescobar please add a line of comment explaining the patch, and add yourself as an author?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9db0217b573939230e7bb8ac984587a8 for a full test report.

**Comment by boegel** [comment]
@pescobar please move the easyconfig files down into the right directory...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2160.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/bbf2479b9ecdcad3fd669c4b3b30d4fe for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @pescobar!

**Pull request merged**